### PR TITLE
Update github/docker/docker to a consistent version across all packages

### PR DESCRIPTION
**Pull request by andrewkroh**
This bumps up the version of all vendored github.com/docker/docker packages to https://github.com/moby/moby/commit/8af4db6f002ac907b6ef8610b237879dfcaa5b7a. Beats was mixing versions of packages from the Docker project.

This does impact netbsd, because this version of Docker doesn't compile for netbsd. If we wanted to address this we could fork moby/moby and apply this commit https://github.com/exekias/moby/commit/83d94aa2b98f10c4534d77077645d5e328533ce4. If we fork then we should also fix the FreeBSD compilation issue referenced in #13400.

I was trying to minimize the amount of changes to the Docker libraries to avoid breaking changes so https://github.com/moby/moby/commit/8af4db6f002ac907b6ef8610b237879dfcaa5b7a is earliest version of Docker that fixes the github.com/Sirupsen/logrus issue causing problems with go.mod in #11850. go.mod also requires that you use the same version for the entire project so this fixes that issue too.

### CODE REVIEW

**Comment by andrewkroh** [comment]
jenkins, test this

File: metricbeat/Makefile [hunk 1]
```diff
@@ -6,7 +6,7 @@ TEST_ENVIRONMENT?=true
 ES_BEATS?=..
 
 # Metricbeat can only be cross-compiled on platforms not requiring CGO.
-GOX_OS=netbsd linux windows
+GOX_OS=linux windows
```
Comment (jsoriano) [inline metricbeat/Makefile]: Umm, why darwin is not here?

**Comment by jsoriano** [comment]
@andrewkroh thanks for taking this!

File: metricbeat/Makefile (hunk 1 above)
Comment (andrewkroh) [reply metricbeat/Makefile]: I think it was not present because it requires CGO and a compiler for darwin.

**Comment by kvch** [comment]
It might be an unpopular opinion, but I don't think we need to support docker (metrics, autodiscovery) on FreeBSD. Docker(moby) does not support it, why should we make an additional effort? I suspect as it's not supported on FreeBSD, not many people use it in production. Forking a lib just puts extra work on us, without any gains.

Instead of forking moby, I would not compile docker related code for FreeBSD (using build tags).

Regarding our PRs, I would take more risks when updating docker. I haven't seen many breaking changes in the latest releases. Also, our automated tests should catch the issues. But I like the changes in you made in our codebase more than in mine. :D So it would be nice to combine both PRs into one. WDYT?

**Comment by andrewkroh** [comment]
> It might be an unpopular opinion, but I don't think we need to support docker (metrics, autodiscovery) on FreeBSD... Instead of forking moby, I would not compile docker related code for FreeBSD (using build tags).

I agree 100%. Let's get those build tags put into place in a separate PR and add back the `netbsd` and `freebsd` cross-compiles. Do you want to try to make this happen?

> Regarding our PRs, I would take more risks when updating docker.

I'm fine with that after seeing the changes in your PR. I didn't want to jump into the unknown 😨 . I'll do an update that pulls the latest docker/docker from docker/engine@<newest-tag>. Then compare it against your PR. I am surprised at how much the two PRs different w.r.t. number of lines changes in vendor/.

**Comment by andrewkroh** [comment]
@kvch I pushed an update with the lastest release from github.com/docker/engine. Let's see how testing goes.

**Comment by andrewkroh** [comment]
There's one test failure happening on Windows for Metricbeat:

> === RUN   TestData
> --- FAIL: TestData (0.00s)
>    modules.go:95: failed to create new MetricSet 1 error: protocol not available
> FAIL
> FAIL	github.com/elastic/beats/metricbeat/module/docker/image	0.265s

**Comment by andrewkroh** [comment]
I think an `integration` build tag was missing because that test should never have been run on Windows. The test is configured to use `unix://` as the host so I'm not surprised it failed. I just wonder why it never failed in the past. I guess Docker improved input validation.

**Comment by andrewkroh** [comment]
jenkins, test this

**Comment by jsoriano** [comment]
+1 to drop docker support on FreeBSD (or any platform not supported by docker itself)

**Comment by andrewkroh** [comment]
jenkins, test this

There was a failure this time in linux/libbeat. Not sure if this was flaky before these changes.

> command [go test -race -tags integration -cover -coverprofile /tmp/gotestcover-545208368 github.com/elastic/beats/libbeat/autodiscover/providers/docker]: exit status 1
> --- FAIL: TestDockerStart (14.03s)
>     docker_integration_test.go:116: Timeout waiting for provider events
> FAIL
> coverage: 76.7% of statements
> FAIL	github.com/elastic/beats/libbeat/autodiscover/providers/docker	14.585s

**Comment by andrewkroh** [comment]
jenkins, test this

That autodiscover test is passing 100% of the time and quickly on machine.

**Comment by andrewkroh** [comment]
The autodiscover test passed in the last run. But Filebeat test_shutdown_wait_ok failed.

**Pull request merged**